### ISSUE TITLE
Automatically extract information from http://minecraft.gamepedia.com/

### ISSUE THREAD

**Issue by rom1504** [PrismarineJS/minecraft-data#8]
There are many ways to extract the data that should go into minecraft-data, as discussed there https://github.com/andrewrk/mineflayer/issues/229 .
In this issue I'll focus on progress on extracting information from the wiki http://minecraft.gamepedia.com/

**Comment by rom1504** [comment]
The current recipes script in the bin/ folder don't produce data to the new recipe format.
But to do that we need the correspondence name -> [id,metadata] ( #7 ) so these scripts can be updated only once we can extract these items/blocks data.

**Comment by rom1504** [comment]
So, in order to do that extraction, I don't want to use html anymore.
 [wikitext](https://en.wikipedia.org/wiki/Help:Wiki_markup) is much more easier to parse.

In order to do get that wikitext there are a few ways : 
* getting the wikitext from the edit page
* using the [API](http://minecraft.gamepedia.com/api.php) : check if it's actually possible to use it
* asking the team in charge of the wiki if they'd be ok with giving [dumps](https://www.mediawiki.org/wiki/Manual:DumpBackup.php) (also see an example here https://dumps.wikimedia.org/) of the database

**Comment by rom1504** [comment]
Using the api is indeed possible ([example](http://minecraft.gamepedia.com/api.php?format=json&action=query&titles=Minecraft_Wiki&prop=revisions&rvprop=content))

Instead of using it manually, let's use this https://github.com/macbre/nodemw

**Comment by rom1504** [comment]
There's also a wikitext parser written in node.js (https://github.com/spencermountain/wtf_wikipedia).
[Parsoid](https://github.com/wikimedia/parsoid) seems like a more advanced parser, but its purpose is generating html so it seems that won't do (https://github.com/spencermountain/wtf_wikipedia/issues/1)

**Comment by rom1504** [comment]
This http://minecraft.gamepedia.com/Data_values is important.

current name in blocks.json and items.json correspond to nothing, wouldn't it be better to replace them by the "nameid" , for example swordDiamond -> diamond_sword (or even minecraft:diamond_sword) ?

**Comment by rom1504** [comment]
http://minecraft.gamepedia.com/Data_values/Block_IDs and http://minecraft.gamepedia.com/Data_values/Item_IDs should be used for the list of blocks and items (that even says if these blocks and items can have metadata) : parsing similar to https://github.com/PrismarineJS/minecraft-data/blob/master/bin/wiki_extractor/entities_extractor.js. Then more data can be found in the page of each block/item.

**Comment by rom1504** [comment]
http://minecraft.gamepedia.com/Category:Documentation_pages useful

**Comment by rom1504** [comment]
Items extraction is done

Now trying blocks extraction : 
* [x] id 
* [x] name
* [x] displayName
* [x] hardness
* [x] stackSize
* [x] diggable
* [x] boundingBox
* [ ] material
* [ ] harvestTools

material goes along with materials.json. Problem is it seems to have been written manually and doesn't correspond to anything specific in the wiki. Most related thing is this http://minecraft.gamepedia.com/Breaking#Best_tools but I don't really know how if it's possible to write materials.json using this.

**Comment by rom1504** [comment]
material : done. materials.json will probably stay manual.

just harvestTools missing.

**Comment by rom1504** [comment]
blocks.json done !

**Comment by rom1504** [comment]
Total progress : 
* [x] entities
* [x] items
* [x] blocks
* [x] materials : manual file (very simple + some edge cases only present in the text on the wiki)
* [ ] biomes
* [ ] instruments
* [ ] recipes

**Comment by rom1504** [comment]
for biomes : http://minecraft.gamepedia.com/Biome + https://github.com/erich666/Mineways/blob/master/Win/biomes.cpp

**Comment by rom1504** [comment]
for recipes : http://minecraft.gamepedia.com/index.php?title=Module:Recipe_list&action=edit

**Comment by rom1504** [comment]
Shapeless means it has multiple shapes. So recipes with only one item, or with 9 times the same item are shaped recipes (see http://minecraft.gamepedia.com/Module_talk:Crafting#Shapeless_recipes_marked_as_shaped_recipes )

**Comment by rom1504** [comment]
recipes done.

**Comment by rom1504** [comment]
only biomes missing.

**Comment by rom1504** [comment]
For biomes : see https://github.com/andrewrk/mineflayer/pull/197

**Comment by rom1504** [comment]
So, biomes current values cannot really be automatically extracted : I added a line in the wiki about how to semi-automatically extract it.

**Comment by rom1504** [comment]
All the .json now have a extraction procedure ! closing.
The next step is about metadata extraction but that will be done in other issues.
